### PR TITLE
Reduce Code Duplication in utils/ical

### LINKED ISSUE #726: Reduce code duplication in utils/ical
There's a bunch of duplicated logic between `utils/ical`, `utils/timify` and `utils/modules`  - mainly small utility functions like `getTimeHour` and `getExamDate`. 

Deduplicating the code should make the code more maintainable and less bug prone.

### PR DESCRIPTION
**Pull request by seanlowjk**
## Context

This fixes #726 which addresses code duplication in the following modules. 
1. `src/utils/ical.ts`
2. `src/utils/timify.ts`
3. `src/utils/modules.ts`

## Implementation

Currently, there exists duplicate code in the 3 modules mentioned above. Such as retrieving the number 
of hours and minutes from `LessonTime` and the retrieval of exam date and duration from a module given the 
current semester. 

As a result, there have been a few new utility functions implemented to address this issue. 

1. `src/utils/timify` 
  a. `getLessonTimeHours` to retrieve the number of hours from `LessonTime` 
  b. `getLessonTimeMinutes` to retrieve the number of minutes from `LessonTime` 
  
2. `src/utils/modules`
  a. `getExamDuration` to retrieve the exam duration of a module given the semester. 

## Other Information

This is as far as to what I have implemented thus far, do let me know if there are any critical 
changes needed to be made, thank you! :smile:

### CODE REVIEW

**Review by ZhangYiJiang — COMMENTED** [review]
This looks pretty good, just some style and test suggestions

File: website/src/utils/timify.ts [hunk 1]
```diff
@@ -14,15 +14,27 @@ import {
 import { TimePeriod } from 'types/venues';
 import { Lesson } from 'types/timetables';
 
+const HOURS_START_INDEX = 0;
+const MINUTES_START_INDEX = 2;
+const DECIMAL_RADIX = 10;
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply website/src/utils/timify.ts:19]: Alright, fair enough!

File: website/src/utils/timify.test.ts [hunk 2]
```diff
@@ -10,8 +10,28 @@ import {
   DEFAULT_EARLIEST_TIME,
   DEFAULT_LATEST_TIME,
   parseDate,
+  getLessonTimeHours,
+  getLessonTimeMinutes,
 } from './timify';
 
+describe('getLessonTimeHours', () => {
+  test('calculates hours correctly', () => {
+    for (let hour = 0; hour < 24; hour += 1) {
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply website/src/utils/timify.test.ts:19]: Alright, I'll stick to using specific test cases as a result.

File: website/src/utils/modules.test.ts [hunk 3]
```diff
@@ -101,11 +102,16 @@ test('formatExamDate should format an exam date string correctly', () => {
...
 
-test('getModuleExamDate should return the correct exam date if it exists', () => {
+test('getExamDate should return the correct exam date if it exists', () => {
   expect(getExamDate(CS1010S, 1)).toBe('2017-11-29T17:00+0800');
   expect(getExamDate(CS3216, 2)).toBeFalsy();
 });
 
+test('getExamDuration should return the correct exam duration if it exists', () => {
+  expect(getExamDuration(CS1010S, 1)).toBe(120);
+  expect(getExamDuration(CS3216, 2)).toBeFalsy();
```
[the remark this answers is not in the document]
Comment (seanlowjk) [reply website/src/utils/modules.test.ts:112]: Ok!

**Comment by seanlowjk** [comment]
Have made the changes as requested, would politely request for your review again thank you! @ZhangYiJiang

**Review by ZhangYiJiang — APPROVED** [review]

**Comment by chrisgzf** [comment]
Thanks for cleaning this up @seanlowjk! Tested on the deploy preview, merging.

**Pull request merged**